lib/bundler/resolver.rb

```diff
@@ -324,8 +324,8 @@ def verify_gemfile_dependencies_are_found!(requirements)
             end
           else
             message = "Could not find gem '#{requirement}' "
-            if @index.source_types.include?(Bundler::Source::Rubygems)
-              message << "in any of the gem sources listed in your Gemfile."
+            if Bundler.definition.has_rubygems_remotes?
+              message << "in any of the gem sources listed in your Gemfile or installed on this machine."
             else
               message << "in the gems available on this machine."
             end
```

Comment:
`@index.source_types` always returns `[Bundler::Index]`, because a source must be an instance `Index` as seen in `Index#add_source`.  So, an instance of `Bundler::Source::Rubygems)` will never be found.